examples/test-exec-doc.yml

```diff
@@ -0,0 +1,221 @@
+# Name of the workflow
+name: Run Tests on Push to Main
+
+# Set permissions required for the workflow
+permissions:
+  contents: write           # Allows writing repository contents
+  pull-requests: write      # Allows creating and modifying pull requests
+  id-token: write           # Allows generating tokens for authentication
+  issues: write             # Allows creating and modifying issues
+
+# Define the events that trigger the workflow
+on:
+  push:
+    paths:
+      - '**/*.md'           # Trigger on push to any Markdown file
+  workflow_dispatch:        # Allows manual triggering of the workflow
+
+jobs:
+  run-tests:
+    runs-on: ubuntu-latest  # Run the job on the latest Ubuntu runner
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 0        # Fetch all history for accurate diff
```

Comment:
I thought we agreed to get rid of this?

Reply:
had to add it back because there were some issues while tracking changed filed between the previous commit and the current one - let's keep it